qiita_db/test/test_artifact.py

```diff
@@ -0,0 +1,427 @@
+# -----------------------------------------------------------------------------
+# Copyright (c) 2014--, The Qiita Development Team.
+#
+# Distributed under the terms of the BSD 3-clause License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# -----------------------------------------------------------------------------
+
+from unittest import TestCase, main
+from tempfile import mkstemp
+from datetime import datetime
+from os import close, remove
+from os.path import exists, join, basename
+from functools import partial
+
+import pandas as pd
+import numpy as np
+from biom.table import Table
+from biom.util import biom_open
+
+from qiita_core.util import qiita_test_checker
+from qiita_db.artifact import Artifact
+from qiita_db.exceptions import (QiitaDBArtifactCreationError,
+                                 QiitaDBArtifactDeletionError,
+                                 QiitaDBOperationNotPermittedError,
+                                 QiitaDBUnknownIDError)
+from qiita_db.metadata_template import PrepTemplate
+from qiita_db.study import Study
+from qiita_db.software import Command, Parameters
+from qiita_db.util import get_mountpoint, get_count
+
+
+@qiita_test_checker()
+class ArtifactTests(TestCase):
+    def setUp(self):
+        # Generate some files for a root artifact
+        fd, self.fp1 = mkstemp(suffix='_seqs.fastq')
+        close(fd)
+        with open(self.fp1, 'w') as f:
+            f.write("@HWI-ST753:189:D1385ACXX:1:1101:1214:1906 1:N:0:\n"
+                    "NACGTAGGGTGCAAGCGTTGTCCGGAATNA\n"
+                    "+\n"
+                    "#1=DDFFFHHHHHJJJJJJJJJJJJGII#0\n")
+
+        fd, self.fp2 = mkstemp(suffix='_barcodes.fastq')
+        close(fd)
+        with open(self.fp2, 'w') as f:
+            f.write("@HWI-ST753:189:D1385ACXX:1:1101:1214:1906 2:N:0:\n"
+                    "NNNCNNNNNNNNN\n"
+                    "+\n"
+                    "#############\n")
+        self.filepaths_root = [(self.fp1, 1), (self.fp2, 3)]
+
+        # Generate some files for a processed artifact
+        fd, self.fp3 = mkstemp(suffix='_seqs.fna')
+        close(fd)
+        with open(self.fp3, 'w') as f:
+            f.write(">1.sid_r4_0 M02034:17:000000000-A5U18:1:1101:15370:1394 "
+                    "1:N:0:1 orig_bc=CATGAGCT new_bc=CATGAGCT bc_diffs=0\n"
+                    "GTGTGCCAGCAGCCGCGGTAATACGTAGGG\n")
+        self.filepaths_processed = [(self.fp3, 4)]
+
+        # Generate some file for a BIOM
+        fd, self.fp4 = mkstemp(suffix='_table.biom')
+        t = Table(np.array([[1, 2], [3, 4]]), ['a', 'b'], ['x', 'y'])
```

Comment:
Pro tip, you can import example_table from biom and you don't have to create a table :D i.e.

```python
from biom import example_table as et

print et
```

No need to change this, but I thought it would be useful to note.